Contribute/how-to-write-use-markdown.md

```diff
@@ -383,10 +383,329 @@ You can see an example of selectors in action at the [Azure docs](https://docs.m
+
+#### Named snippet
+
+Example:
+
+```markdown
+:::code language="csharp" source="intro/samples/cu/Controllers/StudentsController.cs" id="snippet_Create":::
+```
+
+Use only letters and underscores for the name.
+
+The example displays the `snippet_Create` section of the code file. The code file for this example
+has a C# region named `snippet_Create`:
+
+```cs
+// code excluded from the snippet
+// <snippet_Create>
+// code included in the snippet
+// </snippet_Create>
+// code excluded from the snippet
+```
+
+Whenever you can, refer to a named section rather than specifying line numbers. Line number
+references are brittle because code files inevitably change in ways that make line numbers change.
+You don't necessarily get notified of such changes. Your article eventually starts showing the wrong
+lines and you have no clue anything has changed.
+
+#### Highlighting selected lines
+
+Example:
+
+```markdown
+:::code language="csharp" source="intro/samples/cu/Controllers/StudentsController.cs" range="2-24,26" highlight="2,5":::
+```
+
+The example highlights lines 2 and 5, counting from the start of the displayed snippet. (Line
+numbers to highlight don't count from the start of the code file.) In other words, lines 3 and 6
+of the code file are highlighted.
+
+#### Interactive code snippets
+
+You can enable interactive mode for code snippets included by reference. Here are examples:
+
+```markdown
+:::code source="PowerShell.ps1" interactive="cloudshell-powershell":::
+```
+
+```markdown
+:::code source="Bash.sh" interactive="cloudshell-bash":::
+```
+
+To turn on this feature for a particular code block, use the `interactive` attribute. The
+available attribute values are:
+
+- `cloudshell-powershell` - Enables the Azure PowerShell Cloud Shell, as in the preceding
+  example
+- `cloudshell-bash` - Enables the Azure Cloud Shell
+- `try-dotnet` - Enables Try .NET
+- `try-dotnet-class` - Enables Try .NET with class scaffolding
+- `try-dotnet-method` - Enables Try .NET with method scaffolding
+
+For the Azure Cloud Shell and PowerShell Cloud Shell, users can run commands against only their own
+Azure account.
+
+For Try .NET, there are three options for interactivity: `try-dotnet`, `try-dotnet-class`, and `try-dotnet-method`. Use of these options require no additional configuration within the code snippet. The default usings that are being used are:
+
+- System
+- System.Linq
+- System.Collections.Generic
+- System.Text
+- System.Globalization
+- System.Text.RegularExpressions
+
+The `try-dotnet` attribute value enables users to run C# code in the browser without the need to wrap the code in any custom code.
+
+Example:
+
+```md
+:::code source="relative/path/source.cs" interactive="try-dotnet"::
+```
+
+The `try-dotnet-class` value applies class-level scaffolding to the code passed to the interactive component.
+
+```md
+:::code source="relative/path/source.cs" interactive="try-dotnet-class":::
```

Comment:
It would be nice to know what it is best used for vs where it might not.  It is a bit foggy with what is available here, but I get that in this reference style doc we are not loading a lot of detail here.  Maybe link out to those types of details elsewhere, including an overview of the experience it provides?

Reply:
should I link out to this repo? https://github.com/dotnet/try

I feel the README provides a lot of detail. If not, is there somewhere on Docs that provides more info about Try .NET?

Reply:
I think that read me info works great to answer the basic questions of when to use it it and what it will do for you.  Sounds good.

Reply:
Awesome - I'll update this doc and ping you to review. Megan asked that I get your approval before she merges the PR. I'll @ mention you once I'm done!